### ISSUE TITLE
Couldn't convert data to UTF-8 in findMany()

### ISSUE THREAD

**Issue by keli5** [prisma/prisma#20573]
### Bug description

I'm attempting to do a fairly simple findMany for any guildMember with a balance over 0, using the schema and code below, but it fails and throws me this error.

```
0|index  | PrismaClientKnownRequestError:
0|index  | Invalid `prisma.guildMember.findMany()` invocation in
0|index  | C:\Users\letters\Projects\lettersbotnext\commands\leaderboard.js:14:47
0|index  |   11 usage: '[page]',
0|index  |   12 args: false,
0|index  |   13 async execute(client, message, args) {
0|index  | → 14     let people = await prisma.guildMember.findMany(
0|index  | Inconsistent column data: Conversion failed: Couldn't convert data to UTF-8
0|index  |     at Hr.handleRequestError (C:\Users\letters\Projects\lettersbotnext\node_modules\@prisma\client\runtime\library.js:122:6999)
0|index  |     at Hr.handleAndLogRequestError (C:\Users\letters\Projects\lettersbotnext\node_modules\@prisma\client\runtime\library.js:122:6388)
0|index  |     at Hr.request (C:\Users\letters\Projects\lettersbotnext\node_modules\@prisma\client\runtime\library.js:122:6108)
0|index  |     at async l (C:\Users\letters\Projects\lettersbotnext\node_modules\@prisma\client\runtime\library.js:126:10298)
0|index  |     at async Object.execute (C:\Users\letters\Projects\lettersbotnext\commands\leaderboard.js:14:22) {
0|index  |   code: 'P2023',
0|index  |   clientVersion: '5.1.0',
0|index  |   meta: { message: "Conversion failed: Couldn't convert data to UTF-8" }
0|index  | }
``` 

### Expected behavior

Expected it to return an array of guildMembers with balance > 0

### Prisma information

prisma.schema:
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlite"
  url      = env("DATABASE_URL")
}

model Guild {
  id                        String        @id @unique
  joinMesg                  String?
  joinMesgChannel           Int?

  guildMembers              GuildMember[]
  channels                  Channel[]
}

model User {
  id              String        @id @unique
  canUseBot       Boolean       @default(true)
  cooldownExempt  Boolean       @default(false)
  guildMembers    GuildMember[]
}

model GuildMember {
  id      String 
  balance Int       @default(0)
  exp     Int       @default(0)

  globalUser    User   @relation(fields: [globalUserId], references: [id])
  globalUserId  String

  guild         Guild  @relation(fields: [guildId], references: [id])
  guildId       String

  @@id([id, guildId])
}

model Channel {
  id              String    @id @unique
  markovAllowed   Boolean   @default(true)

  guild           Guild     @relation(fields: [guildId], references: [id])
  guildId         String
}
```
index.js:
```js
        let people = await prisma.guildMember.findMany({
            where: {
                balance: {
                    gt: 0
                }
            }
        })
```

### Environment & setup

- OS: Windows 10 22H2
- Database: SQLite
- Node.js version: v20.4.0

### Prisma Version

```
prisma                  : 5.1.0
@prisma/client          : 5.1.0
Current platform        : windows
Query Engine (Node-API) : libquery-engine a9b7003df90aa623086e4d6f4e43c72468e6339b (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli a9b7003df90aa623086e4d6f4e43c72468e6339b (at node_modules\@prisma\engines\schema-engine-windows.exe)
Schema Wasm             : @prisma/prisma-schema-wasm 5.1.0-28.a9b7003df90aa623086e4d6f4e43c72468e6339b
Default Engines Hash    : a9b7003df90aa623086e4d6f4e43c72468e6339b
Studio                  : 0.492.0
```

**Comment by janpio** [comment]
This sounds like Prisma can not handle the data that is being returned for that query. Anything special about it? Can you maybe share it with us? Maybe also use https://pris.ly/d/logging to log the SQL query and run it manually on your database to double check.

**Comment by keli5** [comment]
Here's the database file that I'm having the problem with:
[data.zip](https://github.com/prisma/prisma/files/12269803/data.zip)

**Comment by keli5** [comment]
The query that Prisma generates is:
```
SELECT `main`.`GuildMember`.`id`, `main`.`GuildMember`.`guildId`, `main`.`GuildMember`.`balance`, `main`.`GuildMember`.`exp`, `main`.`GuildMember`.`globalUserId` FROM `main`.`GuildMember` WHERE `main`.`GuildMember`.`balance` > ? LIMIT ? OFFSET ?
```

**Comment by janpio** [comment]
Yes, raw queries with Prisma have the same problem - but seemingly even worse error handling. Sorry.

I downloadded your SQLite database and used "[DB Browser for SQLite](https://sqlitebrowser.org/)" to look into it:
<img width="953" alt="image" src="https://github.com/prisma/prisma/assets/183673/d4697080-bcb1-43b1-999a-a0355633f8ce">
Note how the first row is shown as `BLOB` as `id`. That is unexpected.

The tool has a command "Copy as SQL", and doing that for this field returns: `INSERT INTO "main"."" ("id") VALUES (X'3130353932333530363230393834343434ffff');`. I would guess that is what causes Prisma to crash, as it expects a string - and then gets _something else_. 

Is this value intentional?
If not, can you maybe replace it with a normal id?

Prisma should still have a better error message for this case of course.

**Comment by janpio** [comment]
Just for completeness, I reproduced your error message:
```
> node .\script.js
PrismaClientKnownRequestError: 
Invalid `prisma.guildMember.findMany()` invocation in
C:\Users\Jan\Downloads\data\script.js:6:41

  3 const prisma = new PrismaClient()
  4
  5 async function main() {
→ 6   let people = await prisma.guildMember.findMany(
Inconsistent column data: Conversion failed: Couldn't convert data to UTF-8
    at Hr.handleRequestError (C:\Users\Jan\Downloads\data\node_modules\@prisma\client\runtime\library.js:122:6999)
    at Hr.handleAndLogRequestError (C:\Users\Jan\Downloads\data\node_modules\@prisma\client\runtime\library.js:122:6388)
    at Hr.request (C:\Users\Jan\Downloads\data\node_modules\@prisma\client\runtime\library.js:122:6108)
    at async l (C:\Users\Jan\Downloads\data\node_modules\@prisma\client\runtime\library.js:126:10298)
    at async main (C:\Users\Jan\Downloads\data\script.js:6:16) {
  code: 'P2023',
  clientVersion: '5.1.1',
  meta: { message: "Conversion failed: Couldn't convert data to UTF-8" }
}
```

**Comment by janpio** [comment]
Some further investigation with another tool, [SQLite Viewer](https://marketplace.visualstudio.com/items?itemName=qwtel.sqlite-viewer) for VS Code:
<img width="841" alt="image" src="https://github.com/prisma/prisma/assets/183673/fd966765-d7af-4e8d-8755-0741dabf3274">
Here it shows that the ID looks almost correct - except some weird stuff at the end.

**Comment by janpio** [comment]
Unfortunately I can not interact with the column with the malformed `id` value. I always get `database disk image is malformed` as an error message when I for example try to do a `UPDATE GuildeMember SET id = "123" where balance = 3895` to update the value.

I hope you can just start over, delete the database file and run `db push` and recreate the currently existing data manually or via a seed file.

Do you have any idea how your data(base) got into this state?

**Comment by keli5** [comment]
I've only touched this database file with Prisma, and as far as I can tell, it looks like my mechanical disk may be failing and I just got unlucky. In any case, this is probably not an issue with Prisma querying/handling the database as much as it is error logging.
